test/jasmine/tests/command_test.js

```diff
@@ -0,0 +1,477 @@
+        }]);
+
+        expect(isSimple).toEqual({
+            type: 'data',
+            prop: 'marker.size',
+            traces: null,
+            value: 10
+        });
+    });
+
+    it('return false when properties are not the same', function() {
+        var isSimple = Plots.hasSimpleAPICommandBindings(gd, [{
+            method: 'restyle',
+            args: [{'marker.size': 10}]
+        }, {
+            method: 'restyle',
+            args: [{'marker.color': 20}]
+        }]);
+
+        expect(isSimple).toBe(false);
+    });
+
+    it('return false when a command binds to more than one property', function() {
+        var isSimple = Plots.hasSimpleAPICommandBindings(gd, [{
+            method: 'restyle',
+            args: [{'marker.color': 10, 'marker.size': 12}]
+        }, {
+            method: 'restyle',
+            args: [{'marker.color': 20}]
+        }]);
+
+        expect(isSimple).toBe(false);
+    });
+
+    it('return false when commands affect different traces', function() {
+        var isSimple = Plots.hasSimpleAPICommandBindings(gd, [{
+            method: 'restyle',
+            args: [{'marker.color': 10}, [0]]
+        }, {
+            method: 'restyle',
+            args: [{'marker.color': 20}, [1]]
+        }]);
+
+        expect(isSimple).toBe(false);
+    });
+
+    it('return the binding when commands affect the same traces', function() {
+        var isSimple = Plots.hasSimpleAPICommandBindings(gd, [{
+            method: 'restyle',
+            args: [{'marker.color': 10}, [1]]
+        }, {
+            method: 'restyle',
+            args: [{'marker.color': 20}, [1]]
+        }]);
+
+        expect(isSimple).toEqual({
+            type: 'data',
+            prop: 'marker.color',
+            traces: [ 1 ],
+            value: [ 10 ]
+        });
+    });
+
+    it('return the binding when commands affect the same traces in different order', function() {
+        var isSimple = Plots.hasSimpleAPICommandBindings(gd, [{
+            method: 'restyle',
+            args: [{'marker.color': 10}, [1, 2]]
+        }, {
+            method: 'restyle',
+            args: [{'marker.color': 20}, [2, 1]]
+        }]);
+
+        expect(isSimple).toEqual({
+            type: 'data',
+            prop: 'marker.color',
+            traces: [ 1, 2 ],
+            value: [ 10, 10 ]
+        });
+    });
+});
+
+describe('Plots.computeAPICommandBindings', function() {
+    'use strict';
+
+    var gd;
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+
+        Plotly.plot(gd, [
+            {x: [1, 2, 3], y: [1, 2, 3]},
+            {x: [1, 2, 3], y: [4, 5, 6]},
+        ]);
+    });
+
+    afterEach(function() {
+        destroyGraphDiv(gd);
+    });
+
+    describe('restyle', function() {
+        describe('with invalid notation', function() {
+            it('with a scalar value', function() {
+                var result = Plots.computeAPICommandBindings(gd, 'restyle', [['x']]);
```

Comment:
Thanks for checking this here - as we don't currently don't dive into `args` checking that its items are valid :tada: